src/categorical_algebra/CSets.jl

```diff
@@ -1115,6 +1115,71 @@ function common_ob(A::Subobject, B::Subobject)
   return X
 end
 
+"""
+f:A->B as a map of subobjects of A to subjects of B
+"""
+(f::ACSetTransformation)(X::SubACSet)::SubACSet = begin
+  codom(hom(X)) == dom(f) || error("Cannot apply $f to $X")
+  Subobject(codom(f); Dict(
+    [k=>f.(collect(components(X)[k])) for (k,f) in pairs(components(f))])...)
+end
+
+"""
+f:A->B as a map from A to a subobject of B
+i.e. we cast the ACSet A to its top subobject
+"""
+(f::ACSetTransformation)(X::StructACSet)::SubACSet =
+  X == dom(f) ? f(top(X)) : error("Cannot apply $f to $X")
+
+"""    hom_inv(f::ACSetTransformation,Y::Subobject)::SubACSet
+Inverse of f:A->B as a map of subobjects of B to subjects of A.
+It can be thought of as incident, but for homomorphisms.
+"""
+hom_inv(f::ACSetTransformation,Y::Subobject)::SubACSet = begin
+  codom(hom(Y)) == codom(f) || error("Cannot apply $f to $X")
+
+  Subobject(dom(f); Dict{Symbol, Vector{Int}}(
+    [k => vcat([preimage(f,y) for y in collect(components(Y)[k])]...)
+     for (k,f) in pairs(components(f))])...)
+end
+
+"""    hom_inv(f::CSetTransformation,Y::StructACSet)::SubACSet
+Inverse f:A->B as a map from subobjects of B to subobjects of A.
+Cast an ACSet to subobject, though this has a trivial answer when computing
+the preimage (it is necessarily the top subobject of A).
+"""
+hom_inv(f::CSetTransformation,Y::StructACSet)::SubACSet =
```

Comment:
While we're debugging the docs build, I did notice that this line is apparently not covered by tests. Is this line included in the tests?